bin/provision-instances

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/python3
+#
+# This script runs a full instance provision over all non-ephemeral instances
+# on OCIM. Running this script requires Django to be initialized - the easiest
+# way to do this is by evaluating the script from within the Django shell by
+# running:
+#
+#   > eval(open('bin/provision-instances').open())
+#
+# The provision process happens in "waves", where all instances that fail to
+# provision in a given wave are recorded and reprovisioned during the next wave.
+#
+# This provision script is designed to withstand restarts of either OCIM or
+# this script itself; it accomplishes this by always saving and loading the
+# state of the provision process to/from `ocim_update_waves.ini`.
```

Comment:
Storing this info in the DB is definitely possible, but getting this script done in a trustworthy way on time for rolling out RabbitMQ definitely wouldn't have been possible, especially since that would require migrations that would make it less of an ad-hoc script, and require extensive code review/merging into the code base before use. :)

Keep in mind that this really needed to be a quick external tool to facilitate appserver spawning in order to ensure no spillover. I would have actually much preferred to use an HTTP API to spin up/check on the appservers instead of running `spawn_appserver` directly. In fact, I would have preferred to run this tool on a server external to OCIM altogether, so that it would be able to withstand OCIM outages in a way that wouldn't steer anyone away from restarting/upgrading OCIM.

What are your thoughts about this? Do you think this tool is better suited running alongside OCIM, or do you think we should run it external to OCIM?
I think that either way, we should tease away the need to login to VMs altogether to make ourselves less vulnerable - so if we make this a management command that uses the database for storage, should we also consider having this task be launchable via a web interface instead, and make sure OCIM will resume it in a worker post-restart?

Regardless of the approach, I think we should do it in a separate ticket. :)